tests/spaces/test_discrete.py

```diff
@@ -109,3 +109,56 @@ def test_invalid_probability_mask_values():
         ),
     ):
         space.sample(probability=np.array([0, 0, 0, 0], dtype=np.float64))
+
+
+@pytest.mark.parametrize(
+    "dtype, sample_dtype",
+    [
+        (int, np.int64),
+        (np.int64, np.int64),
+        (np.int32, np.int32),
+        (np.uint8, np.uint8),
+    ],
+)
+def test_dtype(dtype, sample_dtype):
+    space = Discrete(n=5, dtype=dtype, start=2)
+
+    sample = space.sample()
+    sample_mask = space.sample(mask=np.array([0, 1, 0, 0, 0], dtype=np.int8))
+    print(f"{sample=}, {sample_mask=}")
+    print(f"{type(sample)=}, {type(sample_mask)=}")
+    assert isinstance(sample, sample_dtype), type(sample)
+    assert isinstance(sample_mask, sample_dtype), type(sample_mask)
+    # test with probability
+    sample_prob = space.sample(probability=np.array([0.2, 0.2, 0.2, 0.2, 0.2]))
+    assert isinstance(sample_prob, sample_dtype), type(sample_prob)
+
+
+def test_equality():
+
+    space = Discrete(n=3, dtype=np.int32)
+    space_dif_dtype = Discrete(n=3)
+
+    assert space != space_dif_dtype
+
+
+@pytest.mark.parametrize("element", [0, np.int32(0), np.int64(0), np.array(0)])
+@pytest.mark.parametrize("dtype", [np.int32, np.int64, np.int16])
```

Comment:
I'm not sure if this should pass or not for all cases. If you pass a np.int8 to a space with np.int32 that seems ok but the opposite way around I'm not sure